### PR TITLE
Implement Pay ID Get Call with Swagger Codegen

### PR DESCRIPTION
**Pull request by keefertaylor**
## High Level Overview of Change

Implement PayID get using Swagger. 

### Context of Change

Add the same [OpenAPI spec as used in Xpring-JS](https://github.com/xpring-eng/Xpring-JS/blob/master/pay-id-api-spec/pay-id.v1.yml).  Hook up code generator. 

Build the `xrpAddressForPayID` method and wire unit / integration tests. The tests are the same as the [Xpring-JS unit tests](https://github.com/xpring-eng/Xpring-JS/blob/master/test/PayID/pay-id-client-test.ts) and [Xpring-JS integration tests](https://github.com/xpring-eng/Xpring-JS/blob/master/test/PayID/pay-id-integration-test.ts).

### Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (non-breaking change that only restructures code)
- [ ] Tests (You added tests for code that already exists, or your new feature included in this PR)
- [ ] Documentation Updates
- [ ] Release

## Before / After

PayID is now supported.

## Test Plan

New tests provided for CI

### CODE REVIEW

File: src/main/java/io/xpring/payid/PayIDClient.java [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+     *
+     * Note: The returned value will always be in an X-Address format.
+     *
+     * @param payID The payID to resolve for an address.
+     * @return An XRP address representing the given PayID.
+     */
+    public String xrpAddressForPayID(String payID) throws PayIDException {
+        PaymentPointer paymentPointer = PayIDUtils.parsePayID(payID);
+        if (paymentPointer == null) {
+            throw PayIDException.invalidPaymentPointerExpection;
```
Comment (nhartner) [inline src/main/java/io/xpring/payid/PayIDClient.java]: You misspelled _Exception_ here: invalidPaymentPointer**Expection**

File: src/main/java/io/xpring/payid/PayIDClient.java [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+        final String[] localVarAccepts = {
+                "application/xrpl-" + this.network.getNetworkName() + "+json"
+        };
+
+        // NOTE: Swagger produces a higher level client that does not require this level of configuration,
+        // however access to Accept headers is not available unless we access the underlying class.
+        // TODO(keefertaylor): Factor this out to a helper function to make it clear which inputs matter.
+
+        String localVarPath = "/{path}"
+                .replaceAll("\\{" + "path" + "\\}", apiClient.escapeString(path.toString()));
```
Comment (nhartner) [inline src/main/java/io/xpring/payid/PayIDClient.java]: Isn't this equivalent to:
`String localVarPath = "/" +  apiClient.escapeString(path.toString());`

File: src/main/java/io/xpring/payid/PayIDException.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
+package io.xpring.payid;
+
+public class PayIDException extends Exception {
+    /** Static exception for when a classic address is passed to an X-Address API. */
+    public static PayIDException invalidPaymentPointerExpection =
```
Comment (nhartner) [inline src/main/java/io/xpring/payid/PayIDException.java]: fix spelling: invalidPaymentPointerException

File: src/test/java/io/xpring/payid/PayIDClientTest.java [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+import org.junit.rules.ExpectedException;
+
+import static com.github.tomakehurst.wiremock.core.WireMockConfiguration.wireMockConfig;
+import static org.junit.Assert.*;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.*;
+
+public class PayIDClientTest {
+    @Rule
+    public WireMockRule wireMockRule = new WireMockRule(wireMockConfig().dynamicHttpsPort());
```
Comment (nhartner) [inline src/test/java/io/xpring/payid/PayIDClientTest.java]: You will want to make the http port dynamic as well (wiremock starts listening on both).
`new WireMockRule(wireMockConfig().dynamicPort().dynamicHttpsPort());`

File: src/main/java/io/xpring/payid/PayIDClient.java [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+        this.network = network;
+        this.enableSSLVerification = true;
+    }
+
+    /**
+     * Set whether to enable or disable SSL verification.
+     *
+     * Exposed for testing purposes.
+     */
+    public void setEnableSSLVerification(boolean enableSSLVerification) {
```
Comment (nhartner) [inline src/main/java/io/xpring/payid/PayIDClient.java]: Consider annotating this method as `@VisibleForTesting`. It's an annotation that Google uses in their libraries that has become a fairly common convention to use.

File: src/test/java/io/xpring/payid/PayIDClientTest.java (hunk 4 above)
Comment (keefertaylor) [reply src/test/java/io/xpring/payid/PayIDClientTest.java]: PayID only works with HTTPS but I can add this for completeness.

File: src/test/java/io/xpring/payid/PayIDClientTest.java [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+        PayIDClient client = new PayIDClient(XRPLNetwork.MAIN);
+        client.setEnableSSLVerification(false);
+        String xrpAddress = "X7cBcY4bdTTzk3LHmrKAK6GyrirkXfLHGFxzke5zTmYMfw4";
+
+        stubFor(get(urlEqualTo("/georgewashington"))
+                .willReturn(aResponse()
+                        .withStatus(503)));
+
+        // WHEN an XRPAddress is requested THEN a unexpected response error is thrown.
+        // TODO(keefertaylor): Tighten this condition to verify the exception is as expected.
```
[the remark this answers is not in the document]
Comment (keefertaylor) [reply src/test/java/io/xpring/payid/PayIDClientTest.java]: The future, unfortunately. I will be doing a cleanup week sometime after 3/31 and these will all get resolved.

File: src/main/java/io/xpring/payid/PayIDClient.java [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+public class PayIDClient {
+    /**
+     * The network this PayID client resolves on.
+     */
+    private XRPLNetwork network;
+
+    /**
+     * Whether to enable SSL Verification.
+     */
+    private boolean enableSSLVerification;
```
[the remark this answers is not in the document]
Comment (keefertaylor) [reply src/main/java/io/xpring/payid/PayIDClient.java]: Yeah. We use `WireMock` to mock networking. Unfortunately, if we fake HTTPs then Swagger complains we have a bad cert. 

Basically there are two solutions here:
1) Add a cert in the dev / CI environment (ugh)
2) Make it so we can have Swagger not complain in a test env. 

The latter is easier. 

Here's WireMock's documentation:
http://wiremock.org/docs/https/

We use WireMock as it was the library recommended by the ILP team, and I'm trying to keep things consistent.

File: src/main/java/io/xpring/payid/PayIDClient.java (hunk 2 above)
Comment (keefertaylor) [reply src/main/java/io/xpring/payid/PayIDClient.java]: Yeah. I copy / pasted this out of a generated file. Obviously the generated code is more verbose / formulaic. Added your method.

**Pull request merged**